src/app/terms-and-conditions/page.tsx

```diff
@@ -0,0 +1,183 @@
+import { type Metadata } from "next";
+import Link from "next/link";
+
+const Header = ({ children }: { children: React.ReactNode }) => (
+  <h4 className="mb-7 text-xl">{children}</h4>
+);
+
+const Spacer = () => (
+  <span aria-hidden className="my-6 min-h-[1px] min-w-[1px]" />
+);
+
+const TextBlock = ({ children }: { children: React.ReactNode }) => (
+  <div className="flex flex-col justify-start rounded-xl border border-[#212A36] bg-[#0D131C] p-6">
+    {children}
+  </div>
+);
+
+const SmallText = <D extends "r" | "l" | "t" | "b" | "", S extends number>({
+  margins,
+  children,
+}: {
+  margins?:
+    | `m${D}-${S}`
+    | `m${D}-${S} m${D}-${S}`
+    | `m${D}-${S} m${D}-${S} m${D}-${S}`
+    | `m${D}-${S} m${D}-${S} m${D}-${S} m${D}-${S}`;
```

Comment:
This is cool, but a bit over-engineered IMO. Just allow passing `className` with `cn` and don't bother. This doesn't artificially narrow down the control surface over the component while keeping all the benefits. Imagine if tomorrow we also need `paddings` prop